crates/ere-risc0/src/lib.rs

```diff
@@ -232,8 +232,29 @@ impl zkVM for EreRisc0 {
         SDK_VERSION
     }
 
-    fn deserialize_from<R: Read, T: DeserializeOwned>(&self, _reader: R) -> Result<T, zkVMError> {
-        todo!()
+    fn deserialize_from<R: Read, T: DeserializeOwned>(&self, reader: R) -> Result<T, zkVMError> {
+        struct WordReadAdapter<R>(R);
+
+        impl<R: Read> WordRead for WordReadAdapter<R> {
+            fn read_words(&mut self, words: &mut [u32]) -> risc0_zkvm::serde::Result<()> {
+                let bytes = bytemuck::cast_slice_mut(words);
+                self.0
+                    .read_exact(bytes)
+                    .map_err(risc0_zkvm::serde::Error::custom)
+            }
+
+            fn read_padded_bytes(&mut self, bytes: &mut [u8]) -> risc0_zkvm::serde::Result<()> {
+                let mut padded_bytes = vec![0u8; bytes.len().next_multiple_of(4) - bytes.len()];
+                self.0
+                    .read_exact(bytes)
+                    .map_err(risc0_zkvm::serde::Error::custom)?;
+                self.0
+                    .read_exact(&mut padded_bytes)
+                    .map_err(risc0_zkvm::serde::Error::custom)
+            }
+        }
+
+        T::deserialize(&mut Deserializer::new(WordReadAdapter(reader))).map_err(zkVMError::other)
```

Comment:
I was looking at [the docs](https://dev.risczero.com/api/zkvm/tutorials/io#reading-public-data-in-the-host), and they have this [decode()](https://github.com/risc0/risc0/blob/c1ea88f36839b889a15de419e1190ae34c2e0932/risc0/zkvm/src/receipt.rs#L285).. but looks like this is assuming the program only commited as ingle value, and here we want to support multiple commits?

Reply:
Yes, if the guest program only `commit` single thing, we can use the `decode` function, but to support multiple `commit`, we need to have this adapter.